### PR TITLE
1. add version to survey

### PR DESCRIPTION
**Pull request by lxh37**
#17 #24 #32 

1. add version to survey
2. clean and add comments to dynamic survey component
3. save prob, dislike/like to local storage
4. pass prob and current date from dynamic survey component to award-meme and award-altruism component
5. change filename to hash string
6. Add reinforcementData

### CODE REVIEW

File: src/app/storage/aws-s3.service.ts [hunk 1]
```diff
@@ -53,7 +54,7 @@ export class AwsS3Service extends StoreBaseService {
     });  */
 
     //create a file from result passed as a JSONObject
-    var fileName = "result"+new Date().getTime()+".json";
+    var fileName = this.EncrDecr.getSHA256(new Date().getTime())+".json";
```
Comment (mashfiqui-rabbi) [inline src/app/storage/aws-s3.service.ts]: Liying, make this "result_" + (new Date().getTime()) + "_" + this.EncrDecr.getSHA256(username) + ".json"

File: src/app/survey/dynamic-survey/dynamic-survey.component.ts [hunk 2]
```diff
@@ -335,47 +320,33 @@ export class DynamicSurveyComponent implements OnInit {
...
+      console.log("lifeInsightObj: "+JSON.stringify(this.lifeInsightObj));
+      window.localStorage.setItem("lifeInsight", JSON.stringify(this.lifeInsightObj));
+
+      //save to Amazon AWS S3
+      this.awsS3Service.upload(this.fileLink, this.survey2);
+      //console.log("End of storeData");
+
+      //navigate to award-memes/award-altruism with equal probability after submit survey
+      var currentProb = Math.random();
+      window.localStorage.setItem("Prob", ""+currentProb);
```
Comment (mashfiqui-rabbi) [inline src/app/survey/dynamic-survey/dynamic-survey.component.ts]: Liying, we need to change how we handle the probability, like and dislike stuff. We don't want to store "Prob" like this. We want to give Dan a JSON object he can store in the "reinforcement_data." Let us create an object like the following" 

current_reinforcement_data = {"current date": {"probabilility": current_Prob, "reinforcement_type": "meme/alrusim",...} 

Then pass this object to award_memes or award_altruism. Fill in the rest of like or dislike.

Comment (lxh37) [reply src/app/survey/dynamic-survey/dynamic-survey.component.ts]: please let me know the structure of current_reinforcement_data, don't use "...", and I can try.

Comment (mashfiqui-rabbi) [reply src/app/survey/dynamic-survey/dynamic-survey.component.ts]: Liying, Dan knows the JSON structure. Can you ask him for details to make things fast?

But, the earlier structure was the following:

"20170430": {
            "ds": 1, //means participants completed the survey
            "prob": 0.23,
            "Like": "yes", //no if participants hated it.
            "reward": 1,  //0 means users were not randomized
            "reward_type": "meme", //'altruistic message'
            "reward_img_link": "img/reinforcements/memes/IM25.jpg"
}

Comment (lxh37) [reply src/app/survey/dynamic-survey/dynamic-survey.component.ts]: I think it is good to keep the code as it is, most of the information has to be filled in memes /altruism as dynamic-survey doesn't have those info, so this object is better to be created in memes/altriusm than passing from dynamic-survey.

Comment (lxh37) [reply src/app/survey/dynamic-survey/dynamic-survey.component.ts]: I have added reinforcementData and change the file name.

**Comment by mashfiqui-rabbi** [comment]
Thanks, Liying, I merged the pull request. Please pull the latest changes for the latest code.

**Pull request merged**